v2/api/authorization/v1alpha1api20200801previewstorage/structure.txt

```diff
@@ -1,35 +1,38 @@
 github.com/Azure/azure-service-operator/v2/api/authorization/v1alpha1api20200801previewstorage
 ├── APIVersion: Enum (1 value)
 │   └── "2020-08-01-preview"
-└── RoleAssignment: Resource
-    ├── Spec: Object (12 properties)
-    │   ├── AzureName: string
-    │   ├── Condition: *string
-    │   ├── ConditionVersion: *string
-    │   ├── DelegatedManagedIdentityResourceId: *string
-    │   ├── Description: *string
-    │   ├── OriginalVersion: string
-    │   ├── Owner: *genruntime.ArbitraryOwnerReference
-    │   ├── PrincipalId: *string
-    │   ├── PrincipalIdFromConfig: *genruntime.ConfigMapReference
-    │   ├── PrincipalType: *string
-    │   ├── PropertyBag: genruntime.PropertyBag
-    │   └── RoleDefinitionReference: *genruntime.ResourceReference
-    └── Status: Object (17 properties)
-        ├── Condition: *string
-        ├── ConditionVersion: *string
-        ├── Conditions: conditions.Condition[]
-        ├── CreatedBy: *string
-        ├── CreatedOn: *string
-        ├── DelegatedManagedIdentityResourceId: *string
-        ├── Description: *string
-        ├── Id: *string
-        ├── Name: *string
-        ├── PrincipalId: *string
-        ├── PrincipalType: *string
-        ├── PropertyBag: genruntime.PropertyBag
-        ├── RoleDefinitionId: *string
-        ├── Scope: *string
-        ├── Type: *string
-        ├── UpdatedBy: *string
-        └── UpdatedOn: *string
+├── RoleAssignment: Resource
+│   ├── Spec: Object (12 properties)
+│   │   ├── AzureName: string
+│   │   ├── Condition: *string
+│   │   ├── ConditionVersion: *string
+│   │   ├── DelegatedManagedIdentityResourceId: *string
+│   │   ├── Description: *string
+│   │   ├── OriginalVersion: string
+│   │   ├── Owner: *genruntime.ArbitraryOwnerReference
+│   │   ├── PrincipalId: *string
+│   │   ├── PrincipalIdFromConfig: *genruntime.ConfigMapReference
+│   │   ├── PrincipalType: *string
+│   │   ├── PropertyBag: genruntime.PropertyBag
+│   │   └── RoleDefinitionReference: *genruntime.ResourceReference
+│   └── Status: Object (17 properties)
+│       ├── Condition: *string
+│       ├── ConditionVersion: *string
+│       ├── Conditions: conditions.Condition[]
+│       ├── CreatedBy: *string
+│       ├── CreatedOn: *string
+│       ├── DelegatedManagedIdentityResourceId: *string
+│       ├── Description: *string
+│       ├── Id: *string
+│       ├── Name: *string
+│       ├── PrincipalId: *string
+│       ├── PrincipalType: *string
+│       ├── PropertyBag: genruntime.PropertyBag
+│       ├── RoleDefinitionId: *string
+│       ├── Scope: *string
+│       ├── Type: *string
+│       ├── UpdatedBy: *string
+│       └── UpdatedOn: *string
+├── roleAssignmentOverride: Interface
+├── roleAssignment_STATUSOverride: Interface
```

Comment:
I was originally thinking that I would name these `roleAssignmentOverride_STATUS` but that seems worse than `roleAssignment_STATUSOverride`.

Open to better names though.

Reply:
Not sure I love the suffix proposed here (`Properties`) is confusing because there are lots of types named `XyzProperties`, so we'd end up with `AssignXyzPropertiesProperties` which feels pretty awkward.

We could maybe move to just a prefix though? Some ideas...

- `overrideConversionFor<type>` would result in `overrideConversionForRoleAssignment_STATUS`
- `augmentConversionFor<type>` would result in `augmentConversionForRoleAssignment_STATUS`
- `augmentConversion_<type>` could work too: ``augmentConversion_RoleAssignment_STATUS` -- but it does result in more `_`'s which is maybe unfortunate.

I kinda like: `augmentConversionFor<type>` - thoughts?

Reply:
I like `augmentConversionFor<type>`. 👍

Reply:
Updated